x/did/types/did.go

```diff
@@ -462,9 +507,38 @@ func (didDoc DidDocument) HasRelationship(
 	relationships ...string,
 ) bool {
 	// first check if the controller exists
-	for _, vm := range didDoc.VerificationMethods {
-		if vm.BlockchainAccountID != BlockchainAccountID(signer) {
-			continue
+	for _, vm := range didDoc.VerificationMethod {
+
+		switch k := vm.VerificationMaterial.(type) {
+		case *VerificationMethod_BlockchainAccountID:
+			if BlockchainAccountID(k.BlockchainAccountID) != BlockchainAccountID(signer) {
+				continue
+			}
+		case *VerificationMethod_PublicKeyHex:
```

Comment:
@jgimeno this part is the critical one, it's loading a hex pub key and converting it to a cosmos address